File: BinarySearchTree/14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& s) {
        int n = s.size();
        string startString = s[0];
        string ans = "";

        for (int i = 0; i < startString.size(); i++) {
            for (int j = 1; j < s.size(); j++) {
                string nextStr = s[j];
                if (startString[i] != nextStr[i]) {
                    return ans;
                }
            }
            ans += startString[i];
        }
        return ans;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        ans = strs[0]
        for i in range(1, len(strs)):
            ch = ""
            p = 0
            q = 0
            word = strs[i]
            while p < len(ans) and q < len(word):
                if ans[p] == word[q]:
                    ch += ans[p]
                    p+=1
                    q+=1
                else:
                    break
            ans = ch
        return ans
